meta-mel/lib/bblayers/mel_utils.py

```diff
@@ -182,7 +182,21 @@ def layer_for_file(filename):
 
         items_by_layer = defaultdict(set)
         for recipe in fetch_recipes:
-            fn = depgraph['pn'][recipe]['filename']
+            try:
+                fn = depgraph['pn'][recipe]['filename']
```

Comment:
@kergoth that's something already handled at line 200 after the change. We do that after picking up the fname from the graph.

Reply:
From Slack:

You’re right, misunderstood, but I think the approach you’re using is still a problem. It’s going to find any instance of this recipe in any multiconfig to use, but that’s not going to work, a recipe could be built for many configs and have different sources depending on which config it’s being built for.

We need to get the correct sources for the correct recipe, not just any instance of it we can find. So I think we need to take a step back and figure out why an unprefixed recipe name is showing up if it isn’t actually fetched. Is the info in tdepends wrong or incomplete?

I.e. we should be seeing mc:foo:somerecipe in fetch_recipes if that’s the one we have, we shouldn’t see ‘somerecipe’ in the list if it’s really mc:foo:somerecipe

Reply:
As agreed in a meeting yesterday we'll be merging this as a stop gap solution and come back to it later for fixing the root of the issue.
In the current implementation the information in tdepends isn't really complete because the deprecipe does not cater for the multiconfig tag/prefix so that information isn't there.